src/main/clojure/clara/rules/durability/fressian.clj

```diff
@@ -584,19 +587,32 @@
   (deserialize [_ mem-facts opts]
     
     (with-open [^FressianReader rdr (fres/create-reader in-stream :handlers read-handler-lookup)]
-      (let [{:keys [rulebase-only? base-rulebase]} opts
+      (let [{:keys [rulebase-only? base-rulebase compilation-partition-size]} opts
             
             record-holder (ArrayList.)
             ;; The rulebase should either be given from the base-session or found in
             ;; the restored session-state.
             maybe-base-rulebase (when (and (not rulebase-only?) base-rulebase)
                                   base-rulebase)
+
+            compilation-partition-size (or compilation-partition-size 1250)
```

Comment:
Its pretty arbitrary, originally i had 1k but on testing i was able to load a larger kb with 1.25k. 
Honestly, it could probably be a bit higher, it would just come down to the size of the functions being compiled.

Reply:
might be good to say what 1250 is; bytes?  A comment as to its arbitrary nature would be good as well so the next one through feels better if it needs to be altered.

Reply:
No longer accurate, this has changed